### ISSUE TITLE
Scaffolded project with "tsc --init" + "type": "module" fails to compile due to conflicting module/commonjs 

### ISSUE THREAD

**Issue by JessicaSachs** [cypress-io/cypress#27359]
### Current behavior

![image](https://github.com/cypress-io/cypress/assets/2801156/e686fce4-0ae5-443d-98f7-0aae66df577b)

### Desired behavior

Better error message. Something like the below

```
"The module type in your `tsconfig.json` doesn't match the one in your `package.json`. Either choose `commonjs` or `module`"
```

### Test code to reproduce

```
npm init -y
pnpm install cypress typescript -D
pnpm tsc --init
pnpm cypress open
```

Change package.json to use `"type": "module"` (the default for `tsc` is `commonjs`)

### Cypress Version

Any >= 10. Currently 12.X

**Comment by lmiller1990** [comment]
I wonder what makes pnpm different to npm that breaks this?

In fact, I wonder why a package manager has any bearing at all on this. All pnpm does is install packages, it shouldn't impact actual execution, right? 

Are we doing something deeply wrong -- we really shouldn't need package manager specific code at all.

**Comment by lmiller1990** [comment]
I changed it to `cypress.config.mts` (which requires a patch to Cypress, too) but that seems to work. The whole point of `type: module` is so you don't need to add `mts`, though. It seems Cypress is not respecting `type module` when used with pnpm.

**Comment by lmiller1990** [comment]
This has nothing to do with pnpm, same result with `npm`.

**Comment by JessicaSachs** [comment]
If that's the case why doesn't https://github.com/cypress-io/cypress/tree/develop/system-tests/projects/config-with-ts-module-error catch it?

**Comment by JessicaSachs** [comment]
Mr. @marktnoonan supposed that it was pnpm - apologies for not digging more.

**Comment by lmiller1990** [comment]
You can fix with `tsconfig.json` like this:

```json
{
  "compilerOptions": {
    "module": "ES2015"
  }
}
```

Explanation: https://github.com/cypress-io/cypress/issues/23552#issuecomment-1606338593

Bit of a mess, this really should not be the experience users are greeted with. If you don't have a `tsconfig.json`, TypeScript defaults to `"module": "commonjs"` so `cypress.config.ts` compiles to CJS... but since you've got `"type": "module"`, Node.js uses the ESM loader.

I'm not sure what the correct way to fix this would be - we probably don't want to clobber the default `tsconfig.json` options (or do we? The user is asking for ESM, after all).

I am unsure why https://github.com/cypress-io/cypress/tree/develop/system-tests/projects/config-with-ts-module-error does not catch this.

**Comment by JessicaSachs** [comment]
Ahhh I touched "target" when I did `tsc --init` - thank you. That makes SO much sense.

**Comment by lmiller1990** [comment]
Should we close this? Or maybe pivot to "show a better error message"?

Also the test project you linked is used... to assert the exact error you ran into is shown 😇 
https://github.com/cypress-io/cypress/blob/cffa9d1a1856c01d081666b605fc4b402ebf7a92/packages/launchpad/cypress/e2e/config-files-error-handling.cy.ts#L226

I wonder if the error should say something like "Check the `module` value in your `tsconfig.json`" instead of a fairly useless error message".

**Comment by JessicaSachs** [comment]
Yeah I think "The module version in your tsconfig.json doesn't match the one in your package.json" is a good error message.

**Comment by JessicaSachs** [comment]
I cleaned up stuff and expected behavior. Thanks again @lmiller1990 and @marktnoonan for triaging. I thought I was crazy.

**Comment by lmiller1990** [comment]
Ecosystem is the crazy one, especially the CJS + ESM mess.

**Comment by lmiller1990** [comment]
Actually, this happens outside of Cypress, too. We are just bubbling up what Node.js runs into.

1. `npm install typescript`
2. set `"type": "module"`
3. `npx tsc --init`
4. `echo "export const foo = 'bar'"` > "foo.ts"`
5.  `npx tsc -p .`
6. `node foo.js`

```
file:///Users/lachlanmiller/code/dump/dump/foo.js:2
Object.defineProperty(exports, "__esModule", { value: true });
                      ^

ReferenceError: exports is not defined in ES module scope
This file is being treated as an ES module because it has a '.js' file extension and '/Users/lachlanmiller/code/dump/dump/package.json' contains "type": "module". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.
```

So, I guess this is expected behavior? If we work around it, we'd be going against what would happen in a regular TS project.

**Comment by AtofStryker** [comment]
This will be fixed in Cypress 15 with #31520 

### Cypress `14.3.1`

<img width="1198" alt="Image" src="https://github.com/user-attachments/assets/77d0b451-ddd9-4717-a7bb-503e3f5b6a48" />

### Cypress 15 PreRelease 

<img width="1198" alt="Image" src="https://github.com/user-attachments/assets/8b152731-fa60-43ec-9c7e-f238768a86b9" />

**Comment by AtofStryker** [comment]
I am Going to close this issue as it will be resolved with Cypress 15, which is targeted for early July 2025.
